docs/adr/adr-008-monitoring-and-alerting.md

```diff
@@ -0,0 +1,269 @@
+reflects a condition we've formed that the system will not raise critical
+alerts unless we're highly confident it is broken, whereas warnings are
+appropriate for other situations. We decided that for the email alert system
+to be "broken" it represents a failing that will lead to a user noticeable
+issue unless an engineer intervenes. Since these alerts should represent
+a broken system requiring intervention, it should be suitable to contact
+engineers out-of-hours for them as otherwise users will be experiencing issues.
+
+We decided that we will re-approach our alerting with this mindset and look to
+replace alerts that are vulnerable to reflecting a busy status, or which
+extend beyond the boundaries of the system.
+
+## Status
+
+Pending
+
+## Consequences
+
+We will be changing the approaching to monitoring and alerting across the email
+alert system of applications in order to meet this new alerting criteria.
+
+### Monitoring Email Alert API
+
+#### Sidekiq workers
+
+We will configure critical alerts, that contact engineers out-of-hours, if
+individual Sidekiq jobs fail multiple times for the following workers:
+
+- [DailyDigestInitiatorWorker][]
+- [DigestEmailGenerationWorker][]
+- [ProcessContentChangeWorker][]
+- [ProcessMessageWorker][]
+- [SendEmailWorker][]
+- [WeeklyDigestInitiatorWorker][]
+
+These workers will use the database to store the number of times we have
+attempted to perform an action. An alert will be raised if the quantity of
+failed attempts reaches a threshold (the suggestion is 5 - based on this being
+evidence of consistent failure). To ensure that these jobs actually run we
+will configure the [RecoverLostJobsWorker][] to identify any of these that
+need to run and attempt them, this protects from lost work problems such
+as [Sidekiq crashing][sidekiq-reliability].
+
+We expect [SendEmailWorker][] and [DigestEmailGenerationWorker][] jobs to run
+quickly (durations of up to a few seconds) and know that there can be high
+volumes of them to run (both can regularly reach volumes of > 100,000).
+Therefore, we are concerned about a scenario where these consistently fail
+because it could take a long time for sufficient retries to occur
+for an alert to be raised (as each retry is placed at the back of the queue).
+To reflect this concern we will contact engineers if either of these workers
+has not completed one successful run within a time period despite work
+existing (suggested time period is 30 minutes). The [Sidekiq retry
+cadence][sidekiq-retry] and [recovery window][] for these jobs will need to be
+updated to ensure the jobs are always attempted within the time period, this
+is to ensure we don't alert while work isn't being attempted.
+
+We will contact engineers out-of-hours if we determine that
+[DigestRuns][DigestRun] are not created within a reasonable time frame
+(suggestion is 2 hours). Creating these is a [trivial][digest-run-creation],
```

Comment:
Minor: the wording here is slightly at-odds with the definition in the README, which describes the "batch" combo of the DigestRun object and all the DigestRunSubscribers. It may help to explain why we're describing the creation of DigestRun records separately - they are (trivially) created on a schedule, and we're monitoring the schedule.